instrumentation/action_pack/lib/opentelemetry/instrumentation/action_pack/patches/action_controller/metal.rb

```diff
@@ -13,10 +13,16 @@ module ActionController
           module Metal
             def dispatch(name, request, response)
               rack_span = OpenTelemetry::Instrumentation::Rack.current_span
+              route = rails_route(request)
```

Comment:
@dmathieu looks like `route` is only used within `if rack_span.recording?`, so should this be moved inside that condition?